src/components/scenes/CoinRankingScene.tsx

```diff
@@ -139,9 +142,22 @@ const CoinRankingComponent = (props: Props) => {
   })
 
   React.useEffect(() => {
+    if (lastSceneFiat !== coingeckoFiat) {
+      // Clear cache if we changed the fiat while outside of this scene
+      debugLog(LOG_COINRANK, 'Clearing coinRankingDatas cache')
+      coinRanking.coinRankingDatas = []
+      lastStartIndex.current = 1
+      setDataSize(0)
+      setRequestDataSize(QUERY_PAGE_SIZE)
+    }
+
     return () => {
       mounted.current = false
+
+      // Queries should update this, but just in case:
+      lastSceneFiat = coingeckoFiat
```

Comment:
Why not put this in a `useUnmount` hook?

Reply:
See comment
https://github.com/EdgeApp/edge-react-gui/pull/5544#discussion_r2058954481

We need to pass mounted.current around separately anyway, so might as well fold
the changes into this existing effect.

Reply:
Why I suggest `useUnmount` is to avoid the disabled eslint rule hack.